reactor-netty-core/src/main/java/reactor/netty/channel/ChannelOperations.java

```diff
@@ -133,11 +132,10 @@ protected ChannelOperations(ChannelOperations<INBOUND, OUTBOUND> replaced) {
 	 * @param connection the new {@link Connection} connection
 	 * @param listener the events callback
 	 */
-	@SuppressWarnings("deprecation")
 	public ChannelOperations(Connection connection, ConnectionObserver listener) {
 		this.connection = requireNonNull(connection, "connection");
 		this.listener = requireNonNull(listener, "listener");
-		this.onTerminate = MonoProcessor.create();
+		this.onTerminate = Sinks.empty();
```

Comment:
I think this could be `Sinks.unsafe().empty()`, unless there is a need to add overhead of multi-threaded access detection ?

Reply:
It is guaranteed that this will be invoked in an event loop. I'll change it.

Reply:
done